common/network-common/src/main/java/org/apache/spark/network/sasl/SaslRpcHandler.java

```diff
@@ -155,4 +180,48 @@ public void exceptionCaught(Throwable cause, TransportClient client) {
     delegate.exceptionCaught(cause, client);
   }
 
+  /**
+   * Negotiates AES based on complete {@link SparkSaslServer}. The keys need to be encrypted by
+   * sasl server.
+   */
+  private void negotiateAes(ByteBuffer message, RpcResponseCallback callback) {
+    // receive initial option from client
+    CipherOption cipherOption = CipherOption.decode(Unpooled.wrappedBuffer(message));
+    CipherTransformation transformation = CipherTransformation.fromName(cipherOption.cipherSuite);
+    Properties properties = new Properties();
+    properties.setProperty(ConfigurationKeys.CHIMERA_CRYPTO_SECURE_RANDOM_CLASSES_KEY,
+        JavaSecureRandom.class.getName());
+
+    try {
+      // generate key and iv
+      int keyLen = conf.saslEncryptionAesCipherKeySizeBits();
+      KeyGenerator keyGenerator = KeyGenerator.getInstance("HmacSHA1");
```

Comment:
The use of the KeyGenerator for AES key generating is somewhat wrong. For AES, it should,

KeyGenerator.getInstance("AES"); 
instead of 
KeyGenerator.getInstance("HmacSHA1"); 

How about use our SecureRandom  for both generating the AES Key and IV which is simple.